### PR TITLE
layers: Use FormatUtils for imageViewFormatReinterpretation

### PR DESCRIPTION
**Pull request by spencer-lunarg**
Closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/2361

Before (when issue was opened 2+ years ago) we didn't have a generated Format Util that could check against each component bit size

Even if it is compressed formats, it will still match correctly

### CODE REVIEW

File: layers/generated/vk_format_utils.cpp [hunk 1]
```diff
@@ -1935,3 +1935,19 @@ bool FormatHasAlpha(VkFormat format) {
     return FormatHasComponentType(format, COMPONENT_TYPE::A);
 }
 
+bool FormatsSameComponentBits(VkFormat format_a, VkFormat format_b) {
+    auto item_a = kVkFormatTable.find(format_a);
+    auto item_b = kVkFormatTable.find(format_b);
+    if (item_a == kVkFormatTable.end() || item_b == kVkFormatTable.end()) {
+        return false;
+    } else if (item_a->second.component_count != item_a->second.component_count) {
```
Comment (arno-lunarg) [inline layers/generated/vk_format_utils.cpp:1943]:
```suggestion
    } else if (item_a->second.component_count != item_b->second.component_count) {
```

File: scripts/format_utils_generator.py [hunk 2]
```diff
@@ -807,6 +808,22 @@ def sizeFunctions(self):
...
 bool FormatHasAlpha(VkFormat format) {
     return FormatHasComponentType(format, COMPONENT_TYPE::A);
+}
+
+bool FormatsSameComponentBits(VkFormat format_a, VkFormat format_b) {
+    auto item_a = kVkFormatTable.find(format_a);
+    auto item_b = kVkFormatTable.find(format_b);
+    if (item_a == kVkFormatTable.end() || item_b == kVkFormatTable.end()) {
+        return false;
+    } else if (item_a->second.component_count != item_a->second.component_count) {
```
Comment (arno-lunarg) [inline scripts/format_utils_generator.py:818]:
```suggestion
    } else if (item_a->second.component_count != item_b->second.component_count) {
```

**Review by arno-lunarg — APPROVED** [review]
LGTM! One needed change

File: scripts/format_utils_generator.py (hunk 2 above)
Comment (spencer-lunarg) [reply scripts/format_utils_generator.py:818]: :sob: thanks for catching this

Comment (arno-lunarg) [reply scripts/format_utils_generator.py]: I have done it so much time in the past, and will do it countless times in the future, so be wary of my PRs ( ͡• ͜ʖ ͡• )

File: layers/generated/vk_format_utils.cpp [hunk 3]
```diff
@@ -1935,3 +1935,19 @@ bool FormatHasAlpha(VkFormat format) {
     return FormatHasComponentType(format, COMPONENT_TYPE::A);
 }
 
+bool FormatsSameComponentBits(VkFormat format_a, VkFormat format_b) {
+    auto item_a = kVkFormatTable.find(format_a);
+    auto item_b = kVkFormatTable.find(format_b);
```
Comment (juan-lunarg) [inline layers/generated/vk_format_utils.cpp:1940]:
```suggestion
    const auto item_a = kVkFormatTable.find(format_a);
    const auto item_b = kVkFormatTable.find(format_b);
```

File: layers/generated/vk_format_utils.cpp [hunk 4]
```diff
@@ -1935,3 +1935,19 @@ bool FormatHasAlpha(VkFormat format) {
...
+        return false;
+    } else if (item_a->second.component_count != item_b->second.component_count) {
+        return false;
+    }
+    for (uint32_t i = 0; i < item_a->second.component_count; i++) {
+        if (item_a->second.components[i].size != item_b->second.components[i].size) {
+            return false;
+        }
+    }
+    return true;
```
Comment (juan-lunarg) [inline layers/generated/vk_format_utils.cpp:1951]: When adding testing for this function test formats that would hit each of these return statements.

File: layers/generated/vk_format_utils.cpp [hunk 5]
```diff
@@ -1935,3 +1935,19 @@ bool FormatHasAlpha(VkFormat format) {
     return FormatHasComponentType(format, COMPONENT_TYPE::A);
 }
 
+bool FormatsSameComponentBits(VkFormat format_a, VkFormat format_b) {
```
Comment (juan-lunarg) [inline layers/generated/vk_format_utils.cpp:1938]: Add testing for `FormatsSameComponentBits` in `tests/positive/layer_utils.cpp`

**Review by juan-lunarg — CHANGES_REQUESTED** [review]
:+1:

File: layers/generated/vk_format_utils.cpp (hunk 5 above)
Comment (spencer-lunarg) [reply layers/generated/vk_format_utils.cpp:1883]: added tests, good call, there were more nasty edge cases that were failing

**Review by juan-lunarg — APPROVED** [review]

File: tests/negative/portability_subset.cpp [hunk 6]
```diff
@@ -174,18 +167,52 @@ TEST_F(VkPortabilitySubsetTest, CreateImageView) {
...
+    ci.format = VK_FORMAT_B10G11R11_UFLOAT_PACK32;
+    // Incorrect swizzling due to portability
     ci.components.r = VK_COMPONENT_SWIZZLE_IDENTITY;
     ci.components.g = VK_COMPONENT_SWIZZLE_IDENTITY;
     ci.components.b = VK_COMPONENT_SWIZZLE_IDENTITY;
     ci.components.a = VK_COMPONENT_SWIZZLE_IDENTITY;
-    ci.format = VK_FORMAT_R5G6B5_UNORM_PACK16;  // Wrong number of components
+    ci.subresourceRange = {VK_IMAGE_ASPECT_COLOR_BIT, 0, 1, 0, 1};
+    // Format might not be supported
+    m_errorMonitor->SetUnexpectedError("VUID-VkImageViewCreateInfo-None-02273");
```
Comment (juan-lunarg) [inline tests/negative/portability_subset.cpp:214]: Can we detect the ImageView format isn't supported and skip instead?

Comment (ncesario-lunarg) [reply tests/negative/portability_subset.cpp:214]: :heavy_plus_sign:

**Review by ncesario-lunarg — APPROVED** [review]
I completely forgot about this issue :)

File: tests/negative/portability_subset.cpp (hunk 6 above)
Comment (spencer-lunarg) [reply tests/negative/portability_subset.cpp]: If we do, this will never be tested, I am not sure what format is supported on MacOS that was designed to hit `VUID-VkImageViewCreateInfo-imageViewFormatReinterpretation-04466` this way

Added a comment to look into ti
